README.md

```diff
@@ -1,2 +1,352 @@
+# Deep Diving 
+
+Before starting, I want to clear some of my notations. I will abide to the conventions of Linux man pages. In those:
+
+`[ ]` means that anything within (any values, information or parameters) are optional. You can chose one, more than one or none at all.
+`< >` means that the element within (again any parameter, information or a value) is mandatory. We require to replace the text in between angle brackets with actual appropriate information.
+`-`means the options. It does not stand by itself, it will be followed by some form of a value ( or information or parameter).
+
+`-` We will not use that one I guess but while on it a little extra does not kill anyone ... It means long options like a more descriptive name of the option name above. It is pretty common and you may stumble upon them anywhere. For example an option `-l` indicating an option as a 'list', will be given `--list` as a long option as well. Both accomplish the same thing. When you use heavily on options, it helps to use long options for visibility and readability for you and others.
+
+`{ }` I am not sure if this is common. This means within are required options that can be in any relation such as mutual exclusion and so on. One of the specified options or some in conjunction to others *must* be present. For example filtring some results with `{ --time $TODAY | --priority 9 }` means "I want the ones which are from today( assume that entries have times associated with them)" **OR** those who are of priority 9". Note that one of these conditions must be fulfilled. If none of them are satisfied, returned result is empty.
+
+
 # Container
-Container in go
+## A subtle deblurring of the concepts
+
+### What is a Container?
+
+Containers are structured files a.k.a. file system layers and configuration files as JSON blobs to resulting with a packaging scheme and runtime isolation. This is a lot to take in. Let us divide and conquer this.
+
+![file system meme...](/images/fs_meme.png)
+---
+**What does structured files or a file system mean?**
+
+![file system meme...](/images/fs_meme_2.jpeg)
+
+
+Information is contained within a file. A directory itself is a file too. The only difference is that it can contain other files/directories. Directories have children in a tabular form while their content is null. Thus their size is 0.
+
+if we `mdls images/` in this repository, we will see the following. `mdls` stands for Metadata List and as its name suggests it is for listing the metadata of the specified **file**.
+
+Metadata of `images/`             |  metadata of `README.md`
+:-------------------------:|:-------------------------:
+![metadata of folder](/images/metadata_of_images.png)  | ![metadata of README.md file](/images/metadata_of_md.png)
```

Comment:
It is actually amazing to see the differences. However, it is a bit hard for a human eye to compare them. It took me 20 seconds to realize their first line was the same entry... 
I thought, the left one was missing some fields as the first entries, compared to the right one. 
If it won't be too difficult, maybe you can open these 2 images side by side, and take a screenshot, and put it here like that. Would be WAY easier to compare then

Reply:
This makes sense. On top of that, I should color same ones and different ones for the sake of the reader I guess. Thank you for this input.

Reply:
I changed it a bit. How is it now?